src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs

```diff
@@ -282,5 +283,14 @@ private void CreateServiceProvider()
             // service provider, ensuring it will be properly disposed with the provider
             _ = _appServices.GetService<IConfiguration>();
         }
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "ServiceDescriptor has RequiresUnreferencedCode because Service and Implementation types could be generic with mis-matching trimming annotations. " +
```

Comment:
```suggestion
            Justification = "AddSingleton has RequiresUnreferencedCode because Service and Implementation types could be generic with mis-matching trimming annotations. " +
```